### PR TITLE
Linear Probe Hash Table Starter Code

### PR DESCRIPTION
**Pull request by thepulkitagarwal**

### CODE REVIEW

File: src/include/storage/page/hash_table_block_page.h [hunk 1]
```diff
@@ -24,8 +24,79 @@ namespace bustub {
 /**
  * Store indexed key and and value together within block page. Supports
  * non-unique keys.
+ *
+ * Block page format (keys are stored in order):
+ *  ----------------------------------------------------------------
+ * | KEY(1) + VALUE(1) | KEY(2) + VALUE(2) | ... | KEY(n) + VALUE(n)
```
Comment (lmwnshn) [inline src/include/storage/page/hash_table_block_page.h]: This is concatenation and not addition right? Maybe say that.

File: src/include/storage/page/hash_table_block_page.h [hunk 2]
```diff
@@ -24,8 +24,79 @@ namespace bustub {
...
+  /**
+   * Attempts to insert a key and value into an index in the block.
+   * The insert is thread safe. It uses compare and swap to claim the index,
+   * and then writes the key and value into the index, and then marks the
+   * index as readable.
+   *
+   * @param bucket_ind index to write the key and value to
+   * @param key key to insert
+   * @param value value to insert
+   * @return If the value is inserted succesfully, it returns true. If the
```
Comment (lmwnshn) [inline src/include/storage/page/hash_table_block_page.h]: succesfully -> successfully

File: test/container/hash_table_test.cpp [hunk 3]
```diff
@@ -16,18 +16,88 @@
...
   DiskManager *disk_manager = new DiskManager("test.db");
   auto *bpm = new BufferPoolManager(50, disk_manager);
 
-  LinearProbeHashTable<int, int, IntComparator> ht("blah", bpm, IntComparator(), 1000);
-  std::vector<int> vals;
-  EXPECT_FALSE(ht.GetValue(nullptr, 0, &vals));
-  disk_manager->ShutDown();
-  delete bpm;
-  delete disk_manager;
+  LinearProbeHashTable<int, int, IntComparator> ht("blah", bpm, IntComparator(), 1000, &murmur3::MurmurHash3_x64_128);
```
Comment (lmwnshn) [inline test/container/hash_table_test.cpp]: Were we going to provide a Hash wrapper class?

File: src/include/container/hash/linear_probe_hash_table.h [hunk 4]
```diff
@@ -38,31 +39,36 @@ class LinearProbeHashTable {
    * @param buffer_pool_manager buffer pool manager to be used
    * @param comparator comparator for keys
    * @param num_buckets initial number of buckets contained by this hash table
+   * @param hash_fn
```
Comment (lmwnshn) [inline src/include/container/hash/linear_probe_hash_table.h]: Maybe tell them that they can use a pointer to murmur hash, either here or in the writeup.

File: src/include/storage/page/hash_table_block_page.h (hunk 1 above)
Comment (wenxuanqiu) [reply src/include/storage/page/hash_table_block_page.h]: I added a comment.

**Pull request merged**